### ISSUE TITLE
Generate a default .gitlab-ci.yml file for new projects

### ISSUE THREAD

**Issue by stanislas-m** [gobuffalo/buffalo#181]
See #36. This is the same case, but with a different provider.

**Comment by bscott** [comment]
@apaganobeleno I was going to work on this but wondering if I should wait for https://github.com/gobuffalo/buffalo/pull/184 to be merged??

**Comment by apaganobeleno** [comment]
@bscott i think waiting for #184 makes sense, i'll keep you posted on its progress.

**Comment by apaganobeleno** [comment]
@bscott #184 was merged, mentioning this in case you want to work on this one.

**Comment by stanislas-m** [comment]
@bscott, do you want to work on this? I can take care of it otherwise.

**Comment by bscott** [comment]
@stanislas-m By all means please, I'll assign to you

**Comment by stanislas-m** [comment]
Here is my current work on this feature:
https://gist.github.com/stanislas-m/9e202a8881aab9932e41b9e4d8742931

This config will setup a runner to execute the tests for the created project, so I think the feature is at the same level as the Travis one.

I have a remaining problem to solve: the way Gitlab CI manages services (the links to the databases, in this case) force the db host in the databases config file to be changed (I can't use 127.0.0.1: for postgresql it will create a « postgres » host, and so on for mysql).

Do you have any idea on this?

**Comment by markbates** [comment]
My first thought is you want to build this against the v0.8.0 branch, since the generators have been completely refactored. 

As to the database settings, not sure I fully understand the problem, but might I recommend looking at the env and envOr helper methods that let you use env variables in the database.yml file.

**Comment by stanislas-m** [comment]
> My first thought is you want to build this against the v0.8.0 branch, since the generators have been completely refactored.

No problem, I've just done the new version.

> As to the database settings, not sure I fully understand the problem, but might I recommend looking at the env and envOr helper methods that let you use env variables in the database.yml file.

My problem is the way the database.yml is generated:

```yaml
test:
  dialect: postgres
  database: buffalo-test_test
  user: postgres
  password: postgres
  host: 127.0.0.1
```

Since the PostgreSQL server is not on 127.0.0.1 in this case (it will be on a dynamic host created by Gitlab), the test will fail. So, we can either:
- ask the user to change the host by hand
- let Buffalo customize it by tweaking Pop
- change Pop template to use `envOr` helper on test case (so we can provide env in the ci script)
- tweak it at build time, with a bash command like `sed` or `awk`

In my opinion, the third option is the best and would not break legacy setups.

**Comment by stanislas-m** [comment]
Closed by #305.
